lib/packwerk/cache_deprecated_references.rb

```diff
@@ -0,0 +1,45 @@
+# typed: true
+# frozen_string_literal: true
+
+require "sorbet-runtime"
+
+require "packwerk/deprecated_references"
+require "packwerk/reference"
+require "packwerk/reference_lister"
+require "packwerk/violation_type"
+
+module Packwerk
+  class CacheDeprecatedReferences
+    extend T::Sig
+    include ReferenceLister
```

Comment:
I think we can actually make this class abstract - it should never be instantiated, just its child classes, right?

We could to that by using sorbet's `abstract!`, which [can be used to prevent a class from being instantiated.](https://sorbet.org/docs/abstract).

Reply:
done